examples/test_util/EclFile.hpp

```diff
@@ -16,26 +16,83 @@
    along with OPM.  If not, see <http://www.gnu.org/licenses/>.
    */
 
+#ifndef ECLFILE_HPP
+#define ECLFILE_HPP
+
+#include <opm/common/ErrorMacros.hpp>
+#include <examples/test_util/data/EclIOdata.hpp>
 
-//#include "EclIO.hpp"
 #include <iostream>
 #include <string>
 #include <fstream>
 #include <vector>
 #include <ctime>
 #include <map>
-
-#include <examples/test_util/data/EclIOdata.hpp>
+#include <unordered_map>
+#include <stdio.h>
 
 namespace EIOD = Opm::ecl;
 
-class EclFile 
+class EclFile
 {
-
 public:
-    
     explicit EclFile(const std::string& filename);
-    
-};  
+    bool formattedInput() { return formatted; }
+
+    void loadData();                            // load all data
```

Comment:
I would make `loadData()` private - and do that on demand when calling `get()`.

Reply:
i don't agree, the loadData functions are useful for preloading purposes. i have however added the on-demand bit.